extensions/s3/src/main/java/io/deephaven/extensions/s3/S3SeekableChannelProvider.java

```diff
@@ -132,19 +148,21 @@ public boolean isCompatibleWith(@NotNull final SeekableChannelContext channelCon
     @Override
     public CompletableOutputStream getOutputStream(@NotNull final URI uri, final int bufferSizeHint) {
         // bufferSizeHint is unused because s3 output stream is buffered internally into parts
-        return new S3CompletableOutputStream(uri, s3AsyncClient, s3Instructions);
+        return new S3CompletableOutputStream(uriSanitizer.apply(uri), s3AsyncClient, s3Instructions);
     }
 
     @Override
-    public Stream<URI> list(@NotNull final URI directory) {
+    public Stream<URI> list(@NotNull URI directory) {
```

Comment:
Do we need to provide a reverse-sanitizer on the way out? Or, will these new URIs always meant to be used in context with _this_ S3SeekableChannelProvider? I wonder if the contract should be that the scheme of `directory` will be the same scheme of the returns...